docs/usage/shoot_pod_autoscaling_best_practices.md

```diff
@@ -0,0 +1,327 @@
+- As a rule of thumb, set `maxAllowed` to your theoretical maximum load, flanked with alerts to detect erroneous run-away usage or the actual nearing of your practical maximum load, so that you can intervene. However, VPA can easily recommend requests larger than what is allocatable on a node, so you must either ensure large enough nodes (Gardener can scale up from zero, in case you like to define a low-priority worker pool with more resources for very large pods) and/or cap VPA's target recommendations using `maxAllowed` at the node allocatable remainder (after daemonset pods) of the largest eligible machine type (may result in under-provisioning resources for a pod). Use your monitoring and check maximum pod usage to decide about the maximum machine type.
+
+## Recommendations in a Box
+
+| Container | When to use | Value |
+|-----------|-------------|-------|
+| Requests | - **Set them (recommended)** unless:<br/>- Do not set requests for QoS `BestEffort`; useful only if pod can be evicted as often as needed **and** pod can pick up where it left off without any penalty | Set `requests` to 95th percentile of the actually observed CPU resp. memory usage in production if you do not use VPA and the 5th percentile if you use VPA (see below) |
+| Limits | - **Avoid them (recommended)** unless:<br/>- Set limits for QoS `Guaranteed`; useful only if pod has strictly static resource requirements<br/>- Set limits if you know the healthy range, but no more than the node allocatable remainder (after daemonset pods) of the largest eligible machine type | Set `limits` to theoretical max load |
+
+| Scaler | When to use | Initial | Minimum | Maximum |
+|--------|-------------|---------|---------|---------|
+| HPA | **Use for pods that support horizontal scaling** | Set initial `replicas` to 5th percentile of the actually observed replica count in production (prefer scaling on usage, not utilization) and make sure to never overwrite it later when controlled by HPA | Set `minReplicas` to 0 (scale-from-zero) if supported or whatever the high availability requirements demand | Set `maxReplicas` to theoretical max load |
```

Comment:
> Set `minReplicas` to 0 (scale-from-zero) if supported or whatever the high availability requirements demand

I do not think 0 should be the recommended value for minReplicas in HPA. It requires the feature gate `HPAScaleToZero` to be enabled (disabled by default) and at least one Object or External metric is configured, [ref](https://github.com/kubernetes/kubernetes/blob/27c6c30905107593845d4b31281f03fc8b39cc9f/staging/src/k8s.io/api/autoscaling/v2/types.go#L57-L59). 

Practically, the recommended minReplicas should be 1, or maybe 2 when HA is required.

Reply:
Let me add more info... I wanted to illustrate options here, because there is not really a recommendation in this case (it depends too much on the workload).

Reply:
I changed it to:
> Set `minReplicas` to 0 (requires [feature gate](https://github.com/kubernetes/kubernetes/blob/4c7960a67c29b7954cccc6c7d77a62133af3484f/pkg/features/kube_features.go#L266-L267) and [custom/external metrics](https://github.com/kubernetes/kubernetes/pull/74526)), to 1 (regular HPA minimum), or whatever the high availability requirements of the workload demand

Reply:
Thank you @vpnachev, I hope, this covers it.